### DISCUSSION TITLE
"No formulae or casks found for..."

### DISCUSSION THREAD

**Discussion by spikedout** [Homebrew/discussions#3081] [Casks]
Hello, so after force deleting a cask, in particular 'nextcloud' one, and also after manually deleting some related to that particular cask files, I get basically that it's not found in repositories. 
So, how do I fix this?

_OS: macOS Catalina_
_brew version: 3.4.1-56-gf767293_

Thanks in advance.

**Comment by SMillerDev** [comment]
What did you do to delete it and what are you running when you get that message?

**Reply by spikedout** [reply]
As i said, I force deleted it and then also removed any related to that cask files. 

I get that msg on both "install" and "search".

**Reply by SMillerDev** [reply]
Okay, let me be more plain. Without knowing the exact commands and their output, I don't think I can help you.

**Reply by spikedout** [reply]
Output of what commands? I don't have the output of the force deletion. I encountered this problem couple weeks ago. 

I'm pretty sure it's happened because of manual deletion of related files to nextcloud cask.

**Reply by gromgit** [reply]
Then what's the output of the following commands?
```
brew config
brew doctor
brew search nextcloud
```

**Reply by spikedout** [reply]
Hello, see below:

> brew config
```
HOMEBREW_VERSION: 3.4.1-67-gb31d8e9
ORIGIN: https://github.com/Homebrew/brew
HEAD: b31d8e929b09562f431f926d1a158753a1df5877
Last commit: 16 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: c7c90c953757fb0ebbf1e0ff9b193da066881442
Core tap last commit: 70 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_EDITOR: /usr/local/bin/micro
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: quad-core 64-bit haswell
Clang: 12.0.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.7-x86_64
CLT: 12.0.0.32.29
Xcode: N/A

```
>brew doctor

```https://bin.disroot.org/?eb3004be3d0d9d33#PBcncSxbjNkW13T92FsydkzoZFQkri579ShrVUrZLxD```

>brew search nextcloud

```Error: No formulae or casks found for "nextcloud".```

**Reply by carlocab** [reply]
You've mangled your core and cask taps pretty badly. Do
```
rm -rf $(brew --repo homebrew/core)
rm -rf $(brew --repo homebrew/cask)
brew tap homebrew/core
brew tap homebrew/cask
```

**Reply by gromgit** [reply]
The `brew doctor` output shows your problem in graphic detail: You've messed up your core _and_ cask taps with a ton of deletes (including the `nextcloud` cask, hence the error message you saw) and inserts. The most straightforward way to fix both is to re-tap them:
```
rm -fr $(brew --repo homebrew/core)
brew tap homebrew/core
brew untap --force homebrew/cask
brew tap homebrew/cask
```

**Reply by spikedout** [reply]
@gromgit I have successfully applied your's suggested commands and seems that everything's good. I have successfully searched and installed the nextcloud cask. 
Very big thanks to everyone who got in touch with me and helped with this issue! 
Wish you guys nothing but the best!
